pkgs/development/libraries/cairo/default.nix

```diff
@@ -1,171 +1,121 @@
-    # Can be removed after 1.18 release
-    (fetchpatch {
-      name = "fix-grayscale-anialias.patch";
-      url = "https://gitlab.freedesktop.org/cairo/cairo/-/commit/4f4d89506f58a64b4829b1bb239bab9e46d63727.diff";
-      hash = "sha256-mbTg67e7APfdELsuMAgXdY3xokWbGtHF7VDD5UyYqKM=";
-    })
-
-  ];
-
   outputs = [ "out" "dev" "devdoc" ];
   outputBin = "dev"; # very small
+
   separateDebugInfo = true;
 
   nativeBuildInputs = [
+    docbook-xsl-nons
+    gtk-doc
+    meson
+    ninja
     pkg-config
+    python3
   ];
 
-  buildInputs = [
-    libiconv
-    libintl
-  ] ++ optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
+  buildInputs = lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
     CoreGraphics
     CoreText
     ApplicationServices
     Carbon
   ]);
 
-  propagatedBuildInputs = [ fontconfig expat freetype pixman zlib libpng ]
-    ++ optionals x11Support [ libXext libXrender ]
-    ++ optionals xcbSupport [ libxcb xcbutil ]
-    ++ optional gobjectSupport glib
-    ++ optional glSupport libGL
-    ; # TODO: maybe liblzo but what would it be for here?
-
-  configureFlags = [
-    "--enable-tee"
-  ] ++ (if stdenv.isDarwin then [
-    "--disable-dependency-tracking"
-    "--enable-quartz"
-    "--enable-quartz-font"
-    "--enable-quartz-image"
-    "--enable-ft"
-  ] else (optional xcbSupport "--enable-xcb"
-    ++ optional glSupport "--enable-gl"
-    ++ optional pdfSupport "--enable-pdf"
-  )) ++ optional (!x11Support) "--disable-xlib";
-
-  preConfigure =
-  # On FreeBSD, `-ldl' doesn't exist.
-    lib.optionalString stdenv.isFreeBSD
-       '' for i in "util/"*"/Makefile.in" boilerplate/Makefile.in
-          do
-            cat "$i" | sed -es/-ldl//g > t
-            mv t "$i"
-          done
-       ''
-    +
-    ''
-    # Work around broken `Requires.private' that prevents Freetype
-    # `-I' flags to be propagated.
-    sed -i "src/cairo.pc.in" \
-        -es'|^Cflags:\(.*\)$|Cflags: \1 -I${freetype.dev}/include/freetype2 -I${freetype.dev}/include|g'
-    substituteInPlace configure --replace strings $STRINGS
-    '';
-
-  enableParallelBuilding = true;
-
-  doCheck = false; # fails
+  propagatedBuildInputs = [
+    fontconfig
+    freetype
+    glib
+    libpng
+    lzo
+    pixman
+    zlib
+  ] ++ lib.optionals x11Support [
+    libX11
+    libXext
+    libXrender
+  ] ++ lib.optionals xcbSupport [
+    libxcb
+  ];
 
-  postInstall = lib.optionalString stdenv.isDarwin glib.flattenInclude;
+  mesonFlags = [
+    "-Dgtk_doc=true"
+    "-Dspectre=disabled" # only useful for tests
+    "-Dsymbol-lookup=disabled"
+    "-Dtests=disabled"
+    (lib.mesonEnable "xlib" x11Support)
+    (lib.mesonEnable "xcb" xcbSupport)
+  ] ++ (
+    # The meson-cc-tests/ipc_rmid_deferred_release.c test program
+    # won't do its job when cross compiling.
+    let
+      crossFile = writeText "cross-file.conf" ''
+        [properties]
+        ipc_rmid_deferred_release = 'false'
```

Comment:
My research suggested that this should be `true` on linux (https://marc.info/?l=cairo&m=135064523003805&w=2), my resulting file being:

```
      crossFile = writeText "cross-file.conf" ''
        [properties]
        ipc_rmid_deferred_release = ${if stdenv.hostPlatform.isLinux then "true" else "false"}
      '';
```

Reply:
I made the same mistake — my Cairo PR also has the expected values for FreeBSD and NetBSD: https://github.com/NixOS/nixpkgs/pull/257301

Reply:
How many Cairo PRs do we have people.

Reply:
Maybe it will help to keep one open as a draft with a comment 😄